### PR TITLE
python3Packages.duct-py: init at 0.6.4

### PR DESCRIPTION
**Pull request by zmitchell**
###### Description of changes

This Python module provides an easier interface for running subprocesses, and handles many platform- and shell-specific gotchas.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zmitchell** [comment]
Can someone tell me what this error means?

```
       error: The option `lib.maintainers.zmitchell.githubID' does not exist. Definition values:
       - In `/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/maintainers/maintainer-list.nix': 10246891
```

This is the GitHub ID of my account, so I don't understand why it's saying that it doesn't exist.

File: pkgs/development/python-modules/duct-py/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+# tests
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/duct-py/default.nix:4]:
```suggestion
```

`x = y+1; #x is now the successor of y`

File: pkgs/development/python-modules/duct-py/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  pythonImportsCheck = [ "duct" ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    "test_full_env"
+  ];
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/duct-py/default.nix:28]: Explain why this test was disabled: "requires networking", "writes to $HOME"

```suggestion
  disabledTests = [
    # <reasons why>
    "test_full_env"
  ];
```

File: pkgs/development/python-modules/duct-py/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  meta = with lib; {
+    description = "A Python library for running child processes";
+    homepage = "git@github.com:oconnor663/duct.py.git";
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/duct-py/default.nix:32]: What??
```suggestion
    homepage = "http://github.com/oconnor663/duct.py";
```

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -18357,6 +18357,13 @@
     githubId = 1108325;
     name = "Théo Zimmermann";
   };
+  zmitchell = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:18360]: Erase your evidences of fixups.
Squash the commits modifying this file into only one.

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -3129,6 +3129,8 @@ self: super: with self; {
 
   duckduckgo-search = callPackage ../development/python-modules/duckduckgo-search { };
 
+  duct-py = callPackage ../development/python-modules/duct-py { };
```
Comment (AndersonTorres) [inline pkgs/top-level/python-packages.nix:3132]: Read the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) guide **before** marking the "_Fits CONTRIBUTING.md_" checkbox.

> 
>   Format the commit messages in the following way:
> 
>     (pkg-name | nixos/<module>): (from -> to | init at version | refactor | etc)
> 
>     (Motivation for change. Link to release notes. Additional information.)
> 
> For consistency, there should not be a period at the end of the commit message's summary line (the first line of the commit message).
> 
> Examples:
> 
>     nginx: init at 2.0.1

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by zmitchell** [comment]
Hadn't refreshed the page before requesting re-review, disregard

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**